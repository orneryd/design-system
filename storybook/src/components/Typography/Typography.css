.flex-container {
  padding: 0;
  margin: 0;
  list-style: none;

  display: -webkit-box;
  display: -moz-box;
  display: -ms-flexbox;
  display: -webkit-flex;
  display: flex;

  -webkit-flex-flow: row wrap;
  flex-direction: row;
  flex-wrap: wrap;
  justify-content: flex-start;

  width: 50%;
  margin: 0 auto;
  /* border: 1px solid blue; */
}

.flex-container > * {
  padding: 10px;
  flex: 1 100%;
}

.section {
  flex: 0 0 100%;
  margin: 1em 0 0 0;
}

.flex-item {
  max-width: 50%;
  padding: 20px 50px 35px;
  overflow-wrap: break-word;
}

@media screen and (max-width: 800px) {
  .flex-container {
    width: 100%;
  }

  .flex-item {
    max-width: 100%;
  }
}

.header {
  font-family: Georgia, Arial, Helvetica, sans-serif;
}

.content {
    font-family: Arial, Helvetica, sans-serif;
}

.big {
  font-size: 80px;
}

.upper {
  text-transform: uppercase;
}

p {
  margin: 0;
  padding: 0;
}

hr.styled {
    padding: 0;
    margin: 0;
    border: 0;
    height: 1px;
    background: #333;
    background-image: linear-gradient(to right, #ccc, #333, #ccc);
}
