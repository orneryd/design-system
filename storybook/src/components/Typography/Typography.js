import React from 'react';
// import PropTypes from 'prop-types';

export default function Typography() {
  return (

    <div className="flex-container">

      <h2 className="section">Font Families</h2>
      <p>Our typefaces were specifically chosen for readability across formats as well as the way theyâ€™re able to portray our collaborative, forward-thinking, optimistic, authentic and expert personality. Use typography to organize information into clear, easy-to-follow hierarchies that helpthe reader navigate our materials.</p>
      <p>&nbsp;</p>

      <hr className="styled" />

      <div className="flex-item header">
        <span className="big">Aa</span>
        <p>Georgia</p>
        <p>&nbsp;</p>
        <p>abcdefghijklmnopqrstuvwxyz 1234567890 <span className="upper">abcdefghijklmnopqrstuvwxyz</span> <em>abcdefghijklmnopqrstuvwxyz</em></p>
      </div>

      <div className="flex-item content">
        <span className="big">Aa</span>
        <p>Arial</p>
        <p>&nbsp;</p>
        <p>abcdefghijklmnopqrstuvwxyz 1234567890 <span className="upper">abcdefghijklmnopqrstuvwxyz</span> <em>abcdefghijklmnopqrstuvwxyz</em></p>
      </div>

      <hr className="styled" />
      <h3 className="section">Typography</h3>
      <p>Our brand typefaces from Website/Digital applications are <b>Georgia</b> and <b>Arial</b>. We use them for most day-to-day communications, such as reports, presentations or memos. These typefaces are standard on most computer operating systems and accessible for everyone. They are also the typefaces we use for digital applications like websites.</p>

      <p>&nbsp;</p>
      <hr className="styled" />
      <p>&nbsp;</p>
      <p className="small">Headings Default Font - Georgia Bold</p>
      <h1>H1 - Lorem Ipsum dolot amet - 36px</h1>
      <h2>H2 - Lorem Ipsum dolot amet - 28px</h2>
      <h3>H3 - Lorem Ipsum dolot amet - 20px</h3>
      <h4>H4 - Lorem Ipsum dolot amet - 18px </h4>
      <h5>H5 - Lorem Ipsum dolot amet - 16px</h5>
      <h6>H6 - Lorem Ipsum dolot amet</h6>

      <p>&nbsp;</p>
      <hr className="styled" />
      <p>&nbsp;</p>
      <p className="small">Body Default Font - Arial 16px</p>
      <p>&nbsp;</p>
      <p>Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.</p>
      <p><em>Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.</em></p>
      <p>&nbsp;</p>
      <hr className="styled" />
      <p>&nbsp;</p>

    </div>

  );
}

Typography.propTypes = {

};
