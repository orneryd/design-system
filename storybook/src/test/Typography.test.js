import React from 'react';
import ReactDOM from 'react-dom';

import Typography from '../components/Typography/Typography';

describe('Typography', () => {

  it('Typography renders correctly', () => {

    const div = document.createElement('div');
    ReactDOM.render(<Typography />, div);

    /* Makes sures it renders correctly */
    expect(div).toMatchSnapshot();

    ReactDOM.unmountComponentAtNode(div);
  });
  
});
