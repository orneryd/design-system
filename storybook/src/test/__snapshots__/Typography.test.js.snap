// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Typography Typography renders correctly 1`] = `
<div>
  <div
    class="flex-container"
  >
    <h2
      class="section"
    >
      Font Families
    </h2>
    <p>
      Our typefaces were specifically chosen for readability across formats as well as the way they’re able to portray our collaborative, forward-thinking, optimistic, authentic and expert personality. Use typography to organize information into clear, easy-to-follow hierarchies that helpthe reader navigate our materials.
    </p>
    <p>
       
    </p>
    <hr
      class="styled"
    />
    <div
      class="flex-item header"
    >
      <span
        class="big"
      >
        Aa
      </span>
      <p>
        Georgia
      </p>
      <p>
         
      </p>
      <p>
        abcdefghijklmnopqrstuvwxyz 1234567890 
        <span
          class="upper"
        >
          abcdefghijklmnopqrstuvwxyz
        </span>
         
        <em>
          abcdefghijklmnopqrstuvwxyz
        </em>
      </p>
    </div>
    <div
      class="flex-item content"
    >
      <span
        class="big"
      >
        Aa
      </span>
      <p>
        Arial
      </p>
      <p>
         
      </p>
      <p>
        abcdefghijklmnopqrstuvwxyz 1234567890 
        <span
          class="upper"
        >
          abcdefghijklmnopqrstuvwxyz
        </span>
         
        <em>
          abcdefghijklmnopqrstuvwxyz
        </em>
      </p>
    </div>
    <hr
      class="styled"
    />
    <h3
      class="section"
    >
      Typography
    </h3>
    <p>
      Our brand typefaces from Website/Digital applications are 
      <b>
        Georgia
      </b>
       and 
      <b>
        Arial
      </b>
      . We use them for most day-to-day communications, such as reports, presentations or memos. These typefaces are standard on most computer operating systems and accessible for everyone. They are also the typefaces we use for digital applications like websites.
    </p>
    <p>
       
    </p>
    <hr
      class="styled"
    />
    <p>
       
    </p>
    <p
      class="small"
    >
      Headings Default Font - Georgia Bold
    </p>
    <h1>
      H1 - Lorem Ipsum dolot amet - 36px
    </h1>
    <h2>
      H2 - Lorem Ipsum dolot amet - 28px
    </h2>
    <h3>
      H3 - Lorem Ipsum dolot amet - 20px
    </h3>
    <h4>
      H4 - Lorem Ipsum dolot amet - 18px 
    </h4>
    <h5>
      H5 - Lorem Ipsum dolot amet - 16px
    </h5>
    <h6>
      H6 - Lorem Ipsum dolot amet
    </h6>
    <p>
       
    </p>
    <hr
      class="styled"
    />
    <p>
       
    </p>
    <p
      class="small"
    >
      Body Default Font - Arial 16px
    </p>
    <p>
       
    </p>
    <p>
      Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
    </p>
    <p>
      <em>
        Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
      </em>
    </p>
    <p>
       
    </p>
    <hr
      class="styled"
    />
    <p>
       
    </p>
  </div>
</div>
`;
